### PR TITLE
Clamp JAX/NumPy CTC loss so confident alignments stay nonnegative

### LINKED ISSUE #23684: JAX/NumPy CTC loss can be slightly negative for confident alignments
## Bug

Keras's JAX and NumPy CTC implementations are copied from Optax's log-space recursion. For a near-certain valid alignment, float32 rounding can make the per-sequence loss slightly **negative** (on the order of `1e-8`). A downstream `log(loss)` balancer then produces NaNs even though the data and model are finite.

This is the same class of bug as [google-deepmind/optax#1771](https://github.com/google-deepmind/optax/issues/1771). TensorFlow/PyTorch backends use their native CTC ops and are not affected.

## Reproduction (JAX backend)

```python
import numpy as np
import keras

labels = np.array([[1]], dtype="int32")
outputs = np.array([[[0.0, 17.0], [0.0, 17.0]]], dtype="float32")
loss = keras.ops.ctc_loss(labels, outputs, np.array([1]), np.array([2]), mask_index=0)
print(loss)  # can be slightly negative
```

Expected: nonnegative loss. Fix: clamp per-sequence loss at 0.

I have a fix with a regression test and will send a PR.

This PR was prepared with AI assistance. I reviewed the JAX/NumPy CTC implementations against the Optax reproduction and the added test.

### PR DESCRIPTION
**Pull request by Anu27n**
## Summary

The JAX and NumPy CTC backends copy Optax's log-space recursion. For a near-certain valid alignment, float32 rounding can make the per-sequence loss slightly negative, which then NaNs a `log(loss)` balancer. Same class of bug as [optax#1771](https://github.com/google-deepmind/optax/issues/1771). TF/PyTorch backends use native CTC and are unaffected.

Clamp per-sequence loss at 0.

Fixes #23684

## Test plan

- [x] `test_ctc_loss_is_nonnegative_for_confident_alignment`
- [x] existing `test_ctc_loss` expected values unchanged (they were already positive)
- [ ] CI `keras/src/ops/nn_test.py` on JAX and NumPy

## Contributor Agreement

- [x] I am a human, and not a bot.
- [x] I will be responsible for responding to review comments in a timely manner.
- [x] I will work with the maintainers to push this PR forward until submission.

## AI assistance

This PR was written with AI assistance. I reviewed the JAX/NumPy CTC implementations against the Optax reproduction and the added regression test.

### CODE REVIEW

**Comment by Anu27n** [comment]
Torch and OpenVINO CPU jobs failed in infrastructure, not in the CTC change: torch hit `connect ETIMEDOUT 34.118.224.1:443` talking to the k8s API, and OpenVINO hit the 2h runner cap. JAX, NumPy, TensorFlow, and JAX+NNX all passed, which are the backends this PR actually touches.

Contributor-agreement boxes are now checked in the PR description.

**Comment by keerthanakadiri** [comment]
Hi @Anu27n , Can you kindly resolve the conflicts? Thank you!
